src/accelerate/scheduler.py

```diff
@@ -52,18 +53,18 @@ def step(self, *args, **kwargs):
         for opt in self.optimizers:
             if opt.step_was_skipped:
                 return
-
-        if self.split_batches:
-            # Split batches -> the training dataloader batch size is not changed so one step per training step
-            self.scheduler.step(*args, **kwargs)
-        else:
-            # Otherwise the training dataloader batch size was multiplied by `num_processes`, so we need to do
-            # num_processes steps per training step
-            num_processes = AcceleratorState().num_processes
-            for _ in range(num_processes):
-                # Special case when using OneCycle and `drop_last` was not used
-                if getattr(self.scheduler, "total_steps", 0) <= self.scheduler.last_epoch:
-                    self.scheduler.step(*args, **kwargs)
+        if self.gradient_state.sync_gradients:
+            if self.split_batches:
+                # Split batches -> the training dataloader batch size is not changed so one step per training step
+                self.scheduler.step(*args, **kwargs)
+            else:
+                # Otherwise the training dataloader batch size was multiplied by `num_processes`, so we need to do
+                # num_processes steps per training step
+                num_processes = AcceleratorState().num_processes
+                for _ in range(num_processes):
+                    # Special case when using OneCycle and `drop_last` was not used
+                    if getattr(self.scheduler, "total_steps", 0) <= self.scheduler.last_epoch:
+                        self.scheduler.step(*args, **kwargs)
```

Comment:
For the scheduler, I'm not entirely sure how to proceed: we step for each process to account for Distributed training, not sure if we should step all the time or just when synchronizing gradients (since the length set for the scheduler probably does not account for gradient accumulation steps).

Reply:
Resolved: we update every time.